File: 0883-projection-area-of-3d-shapes/0883-projection-area-of-3d-shapes.py
class Solution:
    def projectionArea(self, grid: List[List[int]]) -> int:
        
        top = 0
        rows = len(grid)
        cols = len(grid[0])

        for r in range(rows):
            for c in range(cols):
                if grid[r][c] > 0:
                    top += 1
        
        # side view xz

        side = 0
        for row in grid:
            maxx = max(row)
            side += maxx
            print(row)
        
        front = 0
        # max at each column
        for c in range(cols):
            maxx = 0
            for r in range(rows):
                x = grid[r][c]
                if x > maxx:
                    maxx = x
            front += maxx
        
        return front + top + side

File: 0883-projection-area-of-3d-shapes/0883-projection-area-of-3d-shapes.java
class Solution {
    public int projectionArea(int[][] grid) {
        int n = grid.length;

        int front = 0;
        int side = 0;
        int top = 0;

        for (int r = 0; r < n; r++) {
            int rowMax = 0;

            for (int c = 0; c < n; c++) {
                int val = grid[r][c];

                if (val > 0) {
                    top++;
                }
                
                if (val > rowMax) {
                    rowMax = val;
                }
            }
            front += rowMax;
        }

        for (int c = 0; c < n; c++) {
            int colMax = 0;
            for (int r = 0; r < n; r++) {
                int val = grid[r][c];
                if (val > colMax) {
                    colMax = val;
                }

            }
            side += colMax;
        }
        return side + front + top;
    } 
}